src/js/client/common-filter-widget.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * Filter Widget
+ *
+ * A GPII Handlebars component to build a filter widget of the
+ * type you would type in to filter something like a list of API
+ * functions, a table of settings, or anything else really.
+ */
+"use strict";
+
+var gpii  = fluid.registerNamespace("gpii");
+fluid.registerNamespace("gpii.devpmt");
+fluid.registerNamespace("gpii.devpmt.filterWidget");
+
+fluid.defaults("gpii.devpmt.filterWidget", {
+    gradeNames: ["gpii.handlebars.templateAware", "gpii.binder.bindOnCreate", "gpii.binder.bindOnDomChange"],
+    model: {
+        filterText: null
+    },
+    bindings: { // Binding selectors: modelPaths
+        filterInput: "filterText"
+    },
+    selectors: {
+        filterInput: ".filter-input",
+        resetButtonGroup: ".reset-button-group",
+        resetButton: ".reset-button"
+    },
+    templates: {
+        initial: "common-filter-widget"
+    },
+    invokers: {
+        clearFilter: {
+            funcName: "gpii.devpmt.filterWidget.clearFilter",
+            args: ["{that}"]
+        },
+        renderInitialMarkup: {
+            func: "{that}.renderMarkup",
+            args: ["initial", "{that}.options.templates.initial", "{that}.model"]
+        },
+        updateResetButton: {
+            funcName: "gpii.devpmt.filterWidget.updateResetButton",
+            args: ["{that}", "{that}.model.filterText", "{that}.dom.resetButtonGroup"]
+        },
+        watchInputKeys: {
+            funcName: "gpii.devpmt.filterWidget.watchInputKeys",
+            args: ["{that}", "{arguments}.0"]
+        }
+    },
+    listeners: {
+        onMarkupRendered: [
+            {
+                "this": "{that}.dom.filterInput",
+                "method": "keyup",
+                args: ["{that}.watchInputKeys"]
+            },
+            {
+                "this": "{that}.dom.resetButton",
+                "method": "click",
+                args: ["{that}.clearFilter"]
+            },
+            {
+                "func": "{that}.updateResetButton"
+            }
+        ]
+    },
+    modelListeners: {
+        "filterText": {
+            func: "{that}.updateResetButton"
+        }
+    }
+});
+
+/**
+ * Clear Filter. The filter can be cleared typically by pressing the X button
+ * the right of the field, the ESC key, or anything else bound to this function.
+ */
+gpii.devpmt.filterWidget.clearFilter = function (that) {
+    // For some reason just updating the model isn't updating the bound field.
```

Comment:
Can you isolate this issue and report to @the-t-in-rtf for gpii-binder

Reply:
Hi, I was trying to proactively reproduce this on my machine, I was initially unable to run `npm install` or `yarn install` against the branch in its current state:

```
npm ERR! Command failed: git rev-list -n1 GPII-2474
npm ERR! fatal: ambiguous argument 'GPII-2474': unknown revision or path not in the working tree.
npm ERR! Use '--' to separate paths from revisions, like this:
npm ERR! 'git <command> [<revision>...] -- [<file>...]'
```

You should use the latest dev release instead of the (deleted) branch reference, i.e. `1.1.1-dev.20171212T131257Z.51b112f` at time of writing.  I will attempt to reproduce the problem and comment further shortly.

Reply:
I just realised that version includes breaking changes.  Probably you're best of with 1.0.2 for now, which includes that branch.  However, even with that version I can't load the URL mentioned in the README (http://localhost:8080/prefs/alice).  Seems like that needs to be updated to "editprefs"?  I will assume so and continue investigating.

Reply:
OK, I was able to verify that the model relay is working in both directions, at least with version 1.0.2.  If I change the form value onscreen, I can see in the console that the model variable is updated.  I changed the model variable using a console command like:

`pageComponent.requireRenderer.pageComponent.productList.filter.applier.change("filterText", "iOS")`

The form control value changes and the list of products updates.  I also tested the generic preference settings list using the following command:

```
pageComponent.requireRenderer.pageComponent.genericPrefsTable.filter.applier.change("filterText", "Windows")
```
This also worked.  The only filter input that didn't appear to relay anything is the one in the prefsAdjuster.  This makes sense, because unlike the other examples, the prefsAdjuster has no filter component, no selector for the filter, and no model variable.  I'm assuming it's not that simple, but as far as I can see the things that should work do work, and the things that can't work don't.

@sgithens, please comment with more details about what should be working that isn't, happy to help investigate further.

Reply:
@amb26 @the-t-in-rtf Thanks for all the sluthing on this.  After all the library upgrading I've done in the past week, this is issue is no longer appearing and I've removed the work around.